File: game/tv_show/press_5.py
from game_data import GameData
from phone_events import PhoneEvents
from tv_show.attract import Attract
from tv_show.have_luck import HaveLuck
from tv_show.tv_show_resources import TvShowResources
from tv_show.video_state import VideoState


class Press5(VideoState):
    def __init__(self, context: GameData):
        super().__init__(context, TvShowResources.videos_press_5[context.country], True, TvShowResources.audio_press_5[context.country])

    def process_events(self, phone_events: PhoneEvents):
        super().process_events(phone_events)
        if phone_events.hungup:
            return Attract

        if phone_events.press_5:
            return HaveLuck

        return None

    def render(self, screen):
        super().render(screen)

        if self.has_looped:
            screen.blit(TvShowResources.press_5, (0, 0))

File: game-web/src/tvShow/press5.ts
/** Port of game/tv_show/press_5.py */

import { type StateClass } from '../state';
import { TvShowResources } from './tvShowResources';
import { VideoState } from './videoState';
import { Attract } from './attract';
import { HaveLuck } from './haveLuck';
import type { GameData } from '../gameData';
import type { PhoneEvents } from '../phoneEvents';

export class Press5 extends VideoState {
  constructor(context: GameData) {
    super(
      context,
      TvShowResources.videosPress5[context.country],
      true,
      TvShowResources.audioPress5[context.country],
    );
  }

  processEvents(events: PhoneEvents): StateClass | null {
    super.processEvents(events);
    if (events.hungup) return Attract;
    if (events.press_5) return HaveLuck;
    return null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    super.render(ctx);
    if (this.hasLooped) {
      ctx.drawImage(TvShowResources.press5!, 0, 0);
    }
  }
}
